app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt

```diff
@@ -206,17 +207,184 @@ class StateFragmentTest {
+
+  @Test
+  fun testStateFragment_loadExp_secondState_submitWrongAnswerTwice_feedbackAndQuestionVisible() {
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+
+      // Attempt to submit an wrong answer.
+      typeFractionText("1/3")
+      clickSubmitAnswerButton()
+
+      // Attempt to submit an wrong answer twice.
+      typeFractionText("1/9")
+      clickSubmitAnswerButton()
+
+      scrollToViewType(FEEDBACK)
+      onView(withId(R.id.feedback_text_view)).check(matches(isDisplayed()))
+      onView(withId(R.id.fraction_input_interaction_view)).check(matches(isDisplayed()))
+    }
+  }
+
   //  4. Configuration change with typed text (e.g. for numeric or text input) retains that
   //     temporary
   //     text and you can continue with the exploration after rotating.
+
+  @Test
+  @Ignore("Work on issue regarding this test is in progress")
+  fun testStateFragment_loadExp_submitAnswer_changeConfiguration_explorationContinues() {
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+
+      // Attempt to submit an wrong answer.
+      typeFractionText("1/2")
+
+      rotateToLandscape()
+      clickSubmitAnswerButton()
+
+      clickContinueInteractionButton()
+    }
+  }
+
   //  5. Configuration change after submitting the wrong answer to show that the old answer & re-ask
   //     of the question stay the same.
+  @Test
+  fun testStateFragment_loadExp_submitWrongAnswer_changeConfiguration_feedbackAndQuestionVisible() {
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+
+      // Attempt to submit an wrong answer.
+      typeFractionText("1/3")
+      clickSubmitAnswerButton()
+
+      rotateToLandscape()
+
+      scrollToViewType(FEEDBACK)
+      onView(withId(R.id.feedback_text_view)).check(matches(isDisplayed()))
+      onView(withId(R.id.fraction_input_interaction_view)).check(matches(isDisplayed()))
+    }
+  }
+
   //  6. Backward/forward navigation along with configuration changes to verify that you stay on the
   //     navigated state.
+  @Test
+  fun testStateFragment_loadExp_moveForward_changeConfiguration_stayOnTheNavigatedState() {
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+
+      typeFractionText("1/2")
+      clickSubmitAnswerButton()
+      clickContinueNavigationButton()
+
+      rotateToLandscape()
+      scrollToViewType(CONTENT)
+      val htmlResult =
+        "Which bird can sustain flight for long periods of time?"
+      onView(atPositionOnView(R.id.state_recycler_view, 0, R.id.content_text_view)).check(
+        matches(
+          withText(htmlResult)
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testStateFragment_loadExp_moveForward_moveBackward_changeConfiguration_stayOnSameState() {
```

Comment:
testStateFragment_loadExp_moveForward_moveBackward_stayOnSameState
testStateFragment_loadExp_moveFwd_moveBackwd_moveFwd_stayOnSameState
testStateFragment_loadExp_moveFwd_moveFwd_moveBackwd_moveBackwd_stayOnSameState

and their configChange version too

Reply:
Sure, working on that!

Reply:
Done!